realm/realm-library/src/objectServer/java/io/realm/PermissionManager.java

```diff
@@ -80,34 +81,44 @@ protected Cache initialValue() {
     }
 
     private enum RealmType {
-        PERMISSION_REALM("__permission"),
-        MANAGEMENT_REALM("__management");
+        DEFAULT_PERMISSIONS_REALM("__permission", true),
+        PERMISSION_REALM("__permission", false),
+        MANAGEMENT_REALM("__management", false);
 
         private final String name;
+        private final boolean globalRealm;
 
-        RealmType(String realmName) {
+        RealmType(String realmName, boolean globalRealm) {
             this.name = realmName;
+            this.globalRealm = globalRealm;
         }
 
         public String getName() {
             return name;
         }
+
+        public boolean isGlobalRealm() {
+            return globalRealm;
+        }
     }
 
     private final SyncUser user;
 
     // Used to track the lifecycle of the PermissionManager
     private RealmAsyncTask managementRealmOpenTask;
     private RealmAsyncTask permissionRealmOpenTask;
+    private RealmAsyncTask defaultPermissionsRealmOpenTask;
```

Comment:
If it should be singular, `defaultPermissionsRealmConfig` and `defaultPermissionsRealm` should also be singular.